nupic/torch/modules/k_winners.py

```diff
@@ -90,17 +90,26 @@ def __init__(
         self.k = 0
         self.k_inference = 0
 
-        # Boosting related parameters
+        # Boosting related parameters. Put boost_strength in a buffer so that it
+        # is saved in the state_dict. Keep a copy that remains a Python float so
+        # that its value can be accessed in 'if' statements without blocking to
+        # fetch from GPU memory.
         self.register_buffer("boost_strength", torch.tensor(boost_strength,
                                                             dtype=torch.float))
+        self.boost_strength_cpu = boost_strength
```

Comment:
The name `boost_strength_cpu ` implies `boost_strength` tensor is not on the CPU.
Maybe `_boost_strength_cache`, `_boost_strength_value` or something like that without specifying the device